### PR TITLE
BigQuery : max_results changes page size not full list size

### PR DESCRIPTION
**Pull request by tswast**
Also, running a query is not "preferred" if you do want all the data. If you do run a query, you end up reading data from the destination table anyway.

### CODE REVIEW

**Comment by jonparrott** [comment]
Can you just use `islice` without specifying `max_results`?

**Comment by tswast** [comment]
Done.

File: bigquery/cloud-client/snippets.py [hunk 1]
```diff
@@ -124,10 +125,8 @@ def list_rows(dataset_name, table_name, project=None):
     # Reload the table so that the schema is available.
     table.reload()
 
-    # Load at most 25 results. You can change the max_results argument to load
-    # more rows from BigQuery, but note that this can take some time. It's
-    # preferred to use a query.
-    rows = list(table.fetch_data(max_results=25))
+    # Load at most 25 results.
+    rows = list(itertools.islice(table.fetch_data(), 25))
```
Comment (jonparrott) [inline bigquery/cloud-client/snippets.py]: since you're iterating through the rows below, you don't need to `list(...)`.

Comment (tswast) [inline bigquery/cloud-client/snippets.py]: True. Done.

**Pull request merged**